### PR TITLE
Mention attribute mutators by Proxy and refer to mutators.t in roast

### PR DESCRIPTION
**Pull request by ronaldxs**
Hopefully completion of issue #136 that I had to put aside for a while.  Python, Ruby, C# and Delphi all allow 'setters' for attributes that overload the assignment operator so other people from other languages may wonder about the feature.  Patch tries to give just enough to satisfy their curiosity.

Forgot to mention [March 24 IRC discussion](https://irclog.perlgeek.de/perl6/2018-03-24#i_15961290) with @timo when posting.

### CODE REVIEW

**Comment by ronaldxs** [comment]
@JJ please see if my revision addresses your concerns.  Thank you.

File: doc/Language/objects.pod6 [hunk 1]
```diff
@@ -296,6 +296,27 @@ Method names can be resolved at runtime with the C<.""> operator.
     A.new."$name"().say;
     # OUTPUT: «(Any)␤»
 
+The syntax used to update C<$.notes> changed in this section with respect
+to the previous L<#Attributes> section. Instead of:
+
+    $vacation.notes = 'Pack hiking gear and sunglasses!';
+
+we now:
```
Comment (JJ) [inline doc/Language/objects.pod6]: Verb missing here. Maybe "do"?

File: doc/Language/objects.pod6 [hunk 2]
```diff
@@ -296,6 +296,27 @@ Method names can be resolved at runtime with the C<.""> operator.
...
+available to provide a mutable container on return, and the new update
+method is called to set the attribute rather than an assignment.
+Using a method call is the preferred approach to adding computation
+and logic to the update of an attribute.  Many modern languages can
+update an attribute by overloading assignment with a “setter” method.
+While Perl 6 can overload the assignment operator for this purpose
+with a
+L<C<Proxy>|https://github.com/perl6/roast/blob/master/S12-attributes/mutators.t>
+object, overloading assignment to set attributes with complex logic is
+currently discouraged as weaker OO design.
```
Comment (JJ) [inline doc/Language/objects.pod6]: Expand to "object oriented". Maybe add a reference. Where is it discouraged?

File: doc/Language/objects.pod6 [hunk 3]
```diff
@@ -296,6 +296,27 @@ Method names can be resolved at runtime with the C<.""> operator.
...
+
+Overriding the default auto-generated accessor means it is no longer
+available to provide a mutable container on return, and the new update
+method is called to set the attribute rather than an assignment.
+Using a method call is the preferred approach to adding computation
+and logic to the update of an attribute.  Many modern languages can
+update an attribute by overloading assignment with a “setter” method.
+While Perl 6 can overload the assignment operator for this purpose
+with a
+L<C<Proxy>|https://github.com/perl6/roast/blob/master/S12-attributes/mutators.t>
```
Comment (JJ) [inline doc/Language/objects.pod6]: Wouldn't it be better to make it point to a less terse resource, such as the Perl Advent post? (if I remember correctly)

File: doc/Language/objects.pod6 (hunk 1 above)
Comment (ronaldxs) [reply doc/Language/objects.pod6]: Hopefully addressed in latest commit.

File: doc/Language/objects.pod6 (hunk 2 above)
Comment (ronaldxs) [reply doc/Language/objects.pod6]: Hopefully addressed in latest commit.

File: doc/Language/objects.pod6 (hunk 3 above)
Comment (ronaldxs) [reply doc/Language/objects.pod6]: The Advent post had a Proxy attribute mutator example buried in the middle of a long article and really wouldn't be useful here.

**Comment by ronaldxs** [comment]
If no further objection will merge and close issue #136 later today.

**Comment by JJ** [comment]
I love it when 3-digit issues are closed :-)

**Pull request merged**